src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -270,13 +273,26 @@ public static HybridQueryBuilder fromXContent(XContentParser parser) throws IOEx
         if (isClusterOnOrAfterMinReqVersionForPaginationInHybridQuery()) {
             compoundQueryBuilder.paginationDepth(paginationDepth);
         }
+
+        boolean hasInnerHits = false;
         for (QueryBuilder query : queries) {
             if (filter == null) {
                 compoundQueryBuilder.add(query);
             } else {
                 compoundQueryBuilder.add(query.filter(filter));
             }
+
+            // Check if children have inner hits for stats
+            if (hasInnerHits == false) {
```

Comment:
Where are we changing this parameter value?

Reply:
It's set inside the loop, see my other comment for explanation. Basically my thought is we want to increment the stat once if at least one child query has inner hits, which means the hybrid query as a whole is an inner hits hybrid query.